lib/api/controllers/authController.js

```diff
@@ -369,6 +372,44 @@ class AuthController extends BaseController {
 
     return Bluebird.resolve({});
   }
+
+  /**
+   * @param {Request} request
+   */
+  refreshToken(request) {
+    assertIsAuthenticated(this.anonymousId, request);
+
+    // do not refresh an already refreshed token
```

Comment:
why ?

Reply:
Because otherwise, you can generate many tokens during a token's grace period. And then, from each one of these new tokens, you can once again re-generate other tokens.

Neither Benoit nor I were able to correctly evaluate the risk, but we thought that preventing this potential token bomb was not a bad thing.